ProjFS.Mac/PrjFSKext/PrjFSKext/KauthHandler.cpp

```diff
@@ -412,13 +470,21 @@ static bool ShouldHandleVnodeOpEvent(
     }
     
     *vnodeFileFlags = ReadVNodeFileFlags(vnode, context);
-    if (!FileFlagsBitIsSet(*vnodeFileFlags, FileFlags_IsInVirtualizationRoot))
+    bool filedFlaggedAsInVirtualizationRoot = FileFlagsBitIsSet(*vnodeFileFlags, FileFlags_IsInVirtualizationRoot);
+    if (!filedFlaggedAsInVirtualizationRoot)
     {
-        // This vnode is not part of ANY virtualization root, so exit now before doing any more work.
-        // This gives us a cheap way to avoid adding overhead to IO outside of a virtualization root.
-        
-        *kauthResult = KAUTH_RESULT_DEFER;
-        return false;
+        // TODO(Mac): Determine the overhead of making this check for all actions and
+        // vnodes.  If it's too high, this check can be further reduced to the scenarios
+        // that VFSForGit care about.
+        if (!HasParentInVirtualizationRoot(vnode, context))
```

Comment:
Why did you need this change in behaivor?

Reply:
Without this change VFSForGit would not get PreDelete notifications for the index file as it does not have the `FileFlags_IsInVirtualizationRoot` flag set.

In general the `FileFlags_IsInVirtualizationRoot` will only be set for files inside the root that originated as placeholders.  It will not be set for newly created files or files that were renamed into the root.

Reply:
(And I decided to walk up the tree looking for the flag as we still want to get out the way as fast as we can for files we don't care about, and so I didn't want to add the overhead of looking up the `VirtualizationRoot*` here)

Reply:
In the prototype, we detected that a new file was created, and set the `FileFlags_IsInVirtualizationRoot` on it. The point of that flag is to give you one spot to check for this optimization. Though I think @pmj might have some plans to get rid of that flag and just do a vnode lookup instead.

Reply:
If doing that work avoids the need for workarounds that we're going to immediately throw away, it may be worth combining them.